app/services/hyrax/migrator/crosswalk_metadata_service.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal:true
+
+require 'rdf'
+
+module Hyrax
+  module Migrator
+    # Called by the CrosswalkMetadataActor to map OD1 metadata to OD2
+    class CrosswalkMetadataService
+      include RDF
+
+      CONFIG_FILE = 'config/migrator/crosswalk.yml'
+
+      # Given a graph, returns result hash
+      def crosswalk(graph)
+        graph.statements.each do |statement|
+          data = lookup(statement.predicate.to_s)
+          processed_obj = process(data, statement.object)
+          assemble_hash(data, processed_obj)
+        end
+        @result
+      end
+
+      # Given property data and an object, adds them to result hash
+      def assemble_hash(data, object)
+        return if data[:od2_property].blank?
+
+        @result ||= {}
+        if data[:multiple]
+          @result[data[:od2_property].to_sym] ||= []
+          @result[data[:od2_property].to_sym] += [object]
+        else
+          @result[data[:od2_property].to_sym] = object
+        end
+      end
+
+      # Given an OD2 predicate, returns associated property data or nil
+      def lookup(predicate)
+        hash = crosswalk_hash[:crosswalk]
+        result = (hash.select { |k| k[:od2_predicate] == predicate }).first
```

Comment:
it seems worthwhile to use case-insensitive match here

`k[:predicate].casecmp(predicate).zero?`